mri/reconstructors/base.py

```diff
@@ -0,0 +1,155 @@
+    resulting in inverse transform as solution.
+    """
+
+    def __init__(self, fourier_op, linear_op, regularizer_op,
+                 gradient_formulation, grad_class, lips_calc_max_iter,
+                 num_check_lips, lipschitz_cst, verbose, init_gradient_op=True,
+                 **extra_grad_args):
+        self.fourier_op = fourier_op
+        self.linear_op = linear_op
+        self.prox_op = regularizer_op
+        self.gradient_method = gradient_formulation
+        self.grad_class = grad_class
+        self.lipschitz_cst = lipschitz_cst
+        self.lips_calc_max_iter = lips_calc_max_iter
+        self.num_check_lips = num_check_lips
+        self.verbose = verbose
+        self.extra_grad_args = extra_grad_args
+        if regularizer_op is None:
+            warnings.warn("The prox_op is not set. Setting to identity. "
+                          "Note that optimization is just a gradient descent.")
+            self.prox_op = Identity()
+        # If the reconstruction formulation is synthesis,
+        # we send the linear operator as well.
+        if gradient_formulation == 'synthesis':
+            self.extra_grad_args['linear_op'] = self.linear_op
+        if init_gradient_op:
+            self.initialize_gradient_op(**self.extra_grad_args)
+
+    def initialize_gradient_op(self, **extra_args):
+        # Initialize gradient operator and cost operators
+        self.gradient_op = self.grad_class(
+            fourier_op=self.fourier_op,
+            lips_calc_max_iter=self.lips_calc_max_iter,
+            lipschitz_cst=self.lipschitz_cst,
+            num_check_lips=self.num_check_lips,
+            verbose=self.verbose,
+            **extra_args,
+        )
+        self.cost_op = GenericCost(
+            gradient_op=self.gradient_op,
+            prox_op=self.prox_op,
+            verbose=self.verbose >= 20,
+        )
+
+    def reconstruct(self, kspace_data, optimization_alg='pogm',
+                    x_init=None, num_iterations=100, reinit_grad_op=False,
+                    **kwargs):
+        """ This method calculates operator transform.
+        Parameters
+        ----------
+        kspace_data: np.ndarray
+            the acquired value in the Fourier domain.
+            this is y in above equation.
+        optimization_alg: str (optional, default 'pogm')
+            Type of optimization algorithm to use, 'pogm' | 'fista' |
+            'condatvu'
+        x_init: np.ndarray (optional, default None)
+            input initial guess image for reconstruction
+        num_iterations: int (optional, default 100)
+            number of iterations of algorithm
+        """
+        self.gradient_op.obs_data = kspace_data
+        available_algorthms = ["condatvu", "fista", "pogm"]
+        if optimization_alg not in available_algorthms:
+            raise ValueError("The optimization_alg must be one of " +
+                             str(optimization_alg))
```

Comment:
```suggestion
                             str(available_algorthms))
```
In the tests could you assert that this raises ?